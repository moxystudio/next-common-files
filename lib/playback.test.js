'use strict';

const { withPlayback } = require('../index.js');

const options = {
    dev: true,
    isServer: true,
};

const cleanConfig = () => ({
    module: {
        rules: [],
    },
});

it('should configure a unique rule correctly', () => {
    const webpackConfig = withPlayback()().webpack(cleanConfig(), options);

    expect(webpackConfig.module.rules).toHaveLength(1);
    expect(webpackConfig.module.rules[0]).toMatchSnapshot();
});

it('should return a function if next config is a function', () => {
    const nextConfig = {
        webpack: jest.fn(() => 'foo'),
    };

    const webpackConfig = withPlayback()(nextConfig).webpack(cleanConfig(), options);

    expect(nextConfig.webpack).toHaveBeenCalledTimes(1);
    expect(webpackConfig).toBe('foo');
});
