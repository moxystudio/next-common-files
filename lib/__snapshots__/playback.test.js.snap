// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should configure a unique rule correctly 1`] = `
Object {
  "loader": "<PROJECT_ROOT>/node_modules/url-loader/dist/cjs.js",
  "options": Object {
    "emitFile": false,
    "limit": 0,
    "name": "playback/[name].[ext]",
    "outputPath": "static/chunks/media",
    "publicPath": "/_next/static/chunks/media",
  },
  "test": /\\\\\\.\\(mp3\\|flac\\|wav\\|aac\\|ogg\\|oga\\|mp4\\|m4a\\|webm\\|ogv\\)\\$/,
}
`;
